### ISSUE TITLE
Complex, nested cases seem to break the template

### ISSUE THREAD

**Issue by MichalBryxi** [ember-template-lint/ember-template-recast#706]
Discovered as part of work on [ember-template-lint-plugin-tailwindcss](https://gitlab.com/michal-bryxi/open-source/ember-template-lint-plugin-tailwindcss/-/issues/59) and I _think_ the root cause will be this addon. Reason being that the addon mentioned above _should_ not touch anything outside of `class="..."`.

Given template:

```hbs
<div
  class="pt-2 pb-4 {{this.foo}}"
>
  {{#each this.data as |chunks|}}
    {{#each chunks as |chunk|}}
      {{#if (this.shouldShowImage chunk)}}
        <p class="px-4">foo</p>
      {{else}}
        <p>bar</p>
      {{/if}}
    {{/each}}
  {{/each}}
</div>
```

After triggering auto-fix on `class="pt-2 pb-4 {{this.foo}}"` the code produces: 

```hbs
    -   {{#each this.data as |chunks|}}
    +   {{#each this.data as |chunk|{{#each chunks as |chunk|
```

- Curiously enough removing one of the `each` statements or the `if` will "fix" it.
- It seems to be triggered _only_ when there is "something to fix" by any of the linters in the inner most scope. In my example it's `class="px-4"` that triggers it. But I could also trigger it by having the "bar" and enabling `no-bare-strings`.
- Tested on: `ember-template-recast-6.1.1`

**Comment by MichalBryxi** [comment]
I will try to provide reproduction test for this addon, but not really sure where to start.

**Comment by dcyriller** [comment]
I introduced this regression in #95. https://github.com/ember-template-lint/ember-template-recast/pull/707 reverts #95. It is released in 6.1.2

**Comment by MichalBryxi** [comment]
You are a star @dcyriller! Thank you it [does indeed fix the issue](https://gitlab.com/michal-bryxi/open-source/ember-template-lint-plugin-tailwindcss/-/merge_requests/72) for me ⭐
